src/Symfony/Component/Serializer/Mapping/Factory/CompiledClassMetadataFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Mapping\Factory;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ */
+final class CompiledClassMetadataFactory implements ClassMetadataFactoryInterface
+{
+    private $compiledClassMetadata = array();
+
+    /**
+     * @var ClassMetadataFactoryInterface
+     */
+    private $classMetadataFactory;
+
+    public function __construct(array $compiledClassMetadata, ClassMetadataFactoryInterface $classMetadataFactory)
```

Comment:
passing an array rather than a file to be required means that we cannot have a CacheWarmer generating the file content, as the container will need to contain an array (and so it must be generated at compile-time, which won't allow to use the normal metadata factory to build the metadata as it is not usable yet).